src/java/org/apache/cassandra/tools/nodetool/ListSnapshots.java

```diff
@@ -81,18 +82,20 @@ public void execute(NodeProbe probe)
                 {
                     final List<?> value = (List<?>) eachValue;
                     if (includeEphemeral)
-                        table.add(value.toArray(new String[value.size()]));
-                    else
                         table.add(value.subList(0, value.size() - 1).toArray(new String[value.size() - 1]));
+                    else
+                        table.add(value.subList(0, value.size() - 2).toArray(new String[value.size() - 2]));
+
+                    totalTrueDiskSize += (Long) value.get(value.size() - 1);
                 }
             }
             table.printTo(out);
 
-            out.println("\nTotal TrueDiskSpaceUsed: " + FileUtils.stringifyFileSize(trueSnapshotsSize) + '\n');
+            out.println("\nTotal TrueDiskSpaceUsed: " + FileUtils.stringifyFileSize(totalTrueDiskSize) + '\n');
```

Comment:
I'm worried that changing the "definition" of an already defined field may break some users of this call. I understand that this "breaking change" is gonna live in trunk for the time being, but we need to make sure this change is properly documented.

Reply:
What exactly is breaking here? We literally just sum up numbers instead of calling jmx returning that number. It is same figure, we just sum it up from individual snapshots instead of calling jmx for the second time.

Reply:
If we are not changing the meaning of the metric, then please disregard my comment.